aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -724,13 +724,41 @@ Tensor& normal_out_cuda(Tensor& output, double mean, const Tensor& std, Generato
 }
 
 Tensor& normal_out_cuda(Tensor& output, const Tensor& mean, const Tensor& std, Generator* gen) {
+  bool expandable = are_expandable(mean.sizes(), std.sizes());
+  bool empty_output = output.sizes().equals({0});
+
+  if (expandable) {
+    auto shape = at::infer_size(mean.sizes(), std.sizes());
+    TORCH_CHECK(empty_output || output.sizes().equals(shape), "output size is not the same as broadcast size of mean and std");
+    if (empty_output) {
+      at::native::resize_(output, shape);
+    }
+  }
+  else {
+    TORCH_CHECK(mean.numel() == std.numel(), "mean and std should have same number of element when nonexpandable");
```

Comment:
some more detail here would be good.

For example:
- explain the deprecated support for non-broadcastable shapes
- do we print why the tensors weren't broadcastable?  (we usually do for cases where we are actually checking that).
- do we print the number of elements for both tensors?